components/style/properties/longhand/border.mako.rs

```diff
@@ -273,12 +273,70 @@ ${helpers.predefined_type("border-image-width", "BorderImageWidth",
     initial_value="computed::BorderImageSideWidth::one().into()",
     initial_specified_value="specified::BorderImageSideWidth::one().into()",
     spec="https://drafts.csswg.org/css-backgrounds/#border-image-width",
-    animation_value_type="none",
+    animation_value_type="discrete",
     boxed=True)}
 
 ${helpers.predefined_type("border-image-slice", "BorderImageSlice",
     initial_value="computed::NumberOrPercentage::Percentage(computed::Percentage(1.)).into()",
     initial_specified_value="specified::NumberOrPercentage::Percentage(specified::Percentage(1.)).into()",
     spec="https://drafts.csswg.org/css-backgrounds/#border-image-slice",
-    animation_value_type="none",
+    animation_value_type="discrete",
     boxed=True)}
+
+#[cfg(feature = "gecko")] use values::computed::{LengthOrNumber, NumberOrPercentage};
+#[cfg(feature = "gecko")] impl_rect_conversions!(LengthOrNumber);
+#[cfg(feature = "gecko")] impl_rect_conversions!(NumberOrPercentage);
```

Comment:
Couldn't all of this be in gecko.mako.rs with the rest? Feels quite out of place here.